File: mobile-prototype/ios/CarPlay/MobileStateDTO.swift
import Foundation

// Shared DTOs for mapping /api/mobile_state into platform UI states.
struct MobileStateDTO: Decodable {
    let serverTime: String
    let car: CarDTO?
    let target: TargetDTO?
    let route: RouteDTO
    let eta: EtaDTO

    enum CodingKeys: String, CodingKey {
        case serverTime = "server_time"
        case car, target, route, eta
    }
}

struct CarDTO: Decodable {
    let lat: Double?
    let lon: Double?
    let alt: Double?
    let speed: Double?
    let heading: Double?
    let headingValid: Bool?
    let lastUpdate: String?

    enum CodingKeys: String, CodingKey {
        case lat, lon, alt, speed, heading
        case headingValid = "heading_valid"
        case lastUpdate = "last_update"
    }
}

struct TargetDTO: Decodable {
    let callsign: String
    let landing: LandingDTO
    let timeToLanding: String?
    let timeToLandingS: Int?

    enum CodingKeys: String, CodingKey {
        case callsign, landing
        case timeToLanding = "time_to_landing"
        case timeToLandingS = "time_to_landing_s"
    }
}

struct LandingDTO: Decodable {
    let lat: Double?
    let lon: Double?
    let alt: Double?
}

struct RouteDTO: Decodable {
    let geojson: GeoJSONMinimalDTO?
    let distanceM: Double?
    let durationS: Double?
    let provider: String?
    let providerBase: String?
    let updatedAt: String?

    enum CodingKeys: String, CodingKey {
        case geojson, provider
        case distanceM = "distance_m"
        case durationS = "duration_s"
        case providerBase = "provider_base"
        case updatedAt = "updated_at"
    }
}

struct GeoJSONMinimalDTO: Decodable {
    let type: String?
    let geometry: GeoJSONGeometryDTO?
}

struct GeoJSONGeometryDTO: Decodable {
    let type: String?
    let coordinates: [[Double]]?
}

struct EtaDTO: Decodable {
    let routeDurationS: Double?
    let payloadTimeToLandingS: Int?
    let payloadTimeToLanding: String?

    enum CodingKeys: String, CodingKey {
        case routeDurationS = "route_duration_s"
        case payloadTimeToLandingS = "payload_time_to_landing_s"
        case payloadTimeToLanding = "payload_time_to_landing"
    }
}

File: mobile-prototype/android/MobileStateDto.kt
package org.projecthorus.chasemapper.mobile

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonElement

@Serializable
data class MobileStateDto(
    @SerialName("server_time") val serverTime: String,
    val car: CarDto? = null,
    val target: TargetDto? = null,
    val route: RouteDto,
    val eta: EtaDto
)

@Serializable
data class CarDto(
    val lat: Double? = null,
    val lon: Double? = null,
    val alt: Double? = null,
    val speed: Double? = null,
    val heading: Double? = null,
    @SerialName("heading_valid") val headingValid: Boolean? = null,
    @SerialName("last_update") val lastUpdate: String? = null
)

@Serializable
data class TargetDto(
    val callsign: String,
    val landing: LandingDto,
    @SerialName("time_to_landing") val timeToLanding: String? = null,
    @SerialName("time_to_landing_s") val timeToLandingS: Int? = null
)

@Serializable
data class LandingDto(
    val lat: Double? = null,
    val lon: Double? = null,
    val alt: Double? = null
)

@Serializable
data class RouteDto(
    val geojson: JsonElement? = null,
    @SerialName("distance_m") val distanceM: Double? = null,
    @SerialName("duration_s") val durationS: Double? = null,
    val provider: String? = null,
    @SerialName("provider_base") val providerBase: String? = null,
    @SerialName("updated_at") val updatedAt: String? = null
)

@Serializable
data class EtaDto(
    @SerialName("route_duration_s") val routeDurationS: Double? = null,
    @SerialName("payload_time_to_landing_s") val payloadTimeToLandingS: Int? = null,
    @SerialName("payload_time_to_landing") val payloadTimeToLanding: String? = null
)
